From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Zeno Albisser <zeno.albisser@digia.com>
Date: Wed, 23 Apr 2014 17:01:18 +0200
Subject: Adjust gyp defines for eLinux.

- Add Raspberry PI chipset to list of defined platforms.
- Make sure to include sources for xdg support.
- Do not use pkg-config-wrapper, as our sysroot pkg-config
  already returns absolute paths.
- Use openssl for eLinux.
- Use fontconfig and freetype2 for eLinux.

Change-Id: I8dd21c480922f32ac7111ea5582d82c249414caa
Reviewed-by: Pierre Rossi <pierre.rossi@gmail.com>
---
 base/atomicops_internals_arm_gcc.h |  2 +-
 base/base.gyp                      |  2 +-
 base/base.gypi                     | 12 ++++++++++++
 build/linux/system.gyp             |  2 +-
 net/net.gyp                        |  2 +-
 skia/skia_chrome.gypi              |  2 +-
 skia/skia_library.gypi             |  2 +-
 7 files changed, 18 insertions(+), 6 deletions(-)

diff --git a/base/atomicops_internals_arm_gcc.h b/base/atomicops_internals_arm_gcc.h
index 9f4fe2e..1964041 100644
--- a/base/atomicops_internals_arm_gcc.h
+++ b/base/atomicops_internals_arm_gcc.h
@@ -53,7 +53,7 @@ inline void MemoryBarrier() {
 #if defined(__ARM_ARCH_7__) || defined(__ARM_ARCH_7A__) || \
     defined(__ARM_ARCH_7R__) || defined(__ARM_ARCH_7M__) || \
     defined(__ARM_ARCH_6__) || defined(__ARM_ARCH_6J__) || \
-    defined(__ARM_ARCH_6K__) || defined(__ARM_ARCH_6Z__) || \
+    defined(__ARM_ARCH_6K__) || defined(__ARM_ARCH_6Z__) || defined(__ARM_ARCH_6ZK__) || \
     defined(__ARM_ARCH_6KZ__) || defined(__ARM_ARCH_6T2__)
 
 inline Atomic32 NoBarrier_CompareAndSwap(volatile Atomic32* ptr,
diff --git a/base/base.gyp b/base/base.gyp
index 407d49d..b5af9d7 100644
--- a/base/base.gyp
+++ b/base/base.gyp
@@ -35,7 +35,7 @@
         ],
       },
       'conditions': [
-        ['desktop_linux == 1 or chromeos == 1', {
+        ['desktop_linux == 1 or chromeos == 1 or qt_os == "embedded_linux"', {
           'conditions': [
             ['chromeos==1', {
               'sources/': [ ['include', '_chromeos\\.cc$'] ]
diff --git a/base/base.gypi b/base/base.gypi
index 11edab0..45d7180 100644
--- a/base/base.gypi
+++ b/base/base.gypi
@@ -722,6 +722,18 @@
                 'atomicops_internals_x86_gcc.cc',
               ],
           }],
+          # We need to include xdg for embedded linux,
+          # but we indeed do not want atomicops for x86.
+          ['qt_os == "embedded_linux"', {
+              'sources/': [
+                ['include', '^nix/'],
+              ],
+          }],
+          ['use_qt==1 and target_arch=="ia32"', {
+              'sources/': [
+                ['include', 'atomicops_internals_x86_gcc.cc'],
+              ],
+          }],
           ['<(use_glib)==0 or >(nacl_untrusted_build)==1', {
               'sources!': [
                 'message_loop/message_pump_glib.cc',
diff --git a/build/linux/system.gyp b/build/linux/system.gyp
index 1451afb..92c74805b 100644
--- a/build/linux/system.gyp
+++ b/build/linux/system.gyp
@@ -5,7 +5,7 @@
 {
   'variables': {
     'conditions': [
-      ['sysroot!=""', {
+      ['sysroot!="" and qt_os!="embedded_linux"', {
         'pkg-config': './pkg-config-wrapper "<(sysroot)" "<(target_arch)"',
       }, {
         'pkg-config': 'pkg-config'
diff --git a/net/net.gyp b/net/net.gyp
index c2d618e..69f4827 100644
--- a/net/net.gyp
+++ b/net/net.gyp
@@ -1366,7 +1366,7 @@
               '../build/linux/system.gyp:gio',
             ],
         }],
-        [ 'desktop_linux == 1 or chromeos == 1', {
+        [ 'desktop_linux == 1 or chromeos == 1 or qt_os == "embedded_linux"', {
             'conditions': [
               ['use_openssl==1', {
                 'dependencies': [
diff --git a/skia/skia_chrome.gypi b/skia/skia_chrome.gypi
index 6dcc665..5559f57 100644
--- a/skia/skia_chrome.gypi
+++ b/skia/skia_chrome.gypi
@@ -117,7 +117,7 @@
   'target_conditions': [
     # Pull in specific linux files for android (which have been filtered out
     # by file name rules).
-    [ 'OS == "android"', {
+    [ 'OS == "android" or qt_os == "embedded_linux"', {
       'sources/': [
         ['include', 'ext/platform_device_linux\\.cc$'],
       ],
diff --git a/skia/skia_library.gypi b/skia/skia_library.gypi
index b957fa6..effde33 100644
--- a/skia/skia_library.gypi
+++ b/skia/skia_library.gypi
@@ -270,7 +270,7 @@
         '../third_party/skia/src/opts/opts_check_SSE2.cpp'
       ],
     }],
-    [ 'desktop_linux == 1 or chromeos == 1', {
+    [ 'desktop_linux == 1 or chromeos == 1 or qt_os=="embedded_linux"', {
       'dependencies': [
         '../build/linux/system.gyp:fontconfig',
         '../build/linux/system.gyp:freetype2',
